### PR TITLE
remove seeding of to be deprecated PoolCandidate fields

### LINKED ISSUE #4263: Update seeders to not populate deprecated columns on pool_candidate rows.
Issue #3148 is to migrate a bunch of deprecated fields from the pool_candidate object to the user object.  Let's update the seeders to not populate these deprecated fields on pool_candidate so we don't accidentally perform new development on them.

### PR DESCRIPTION
**Pull request by vd1992**
closes #4263 
deletes fields from the factory/seeding that are to be deprecated 

testing:
checks pass
migrating/seeding looks right in adminer

### CODE REVIEW

**Review by JamesHuf — APPROVED** [review]
This looks good to me. I'm thinking it might be a good idea to mark the fields as deprecated in graphql as well (in another issue).

File: api/database/seeders/DatabaseSeeder.php [hunk 1]
```diff
@@ -184,7 +184,6 @@ private function seedPoolCandidate(User $user, Pool $pool)
         PoolCandidate::factory()->count(1)->sequence(fn () => [
             'pool_id' => $pool->id,
             'user_id' => $user->id,
-            'expected_salary' => $user->expected_salary
         ])->for($user)->afterCreating(function (PoolCandidate $candidate) use ($user) {
             // match arrays from the user
             $candidate->expectedClassifications()->sync($user->expectedClassifications()->pluck('classifications.id')->toArray());
```
Comment (petertgiles) [inline api/database/seeders/DatabaseSeeder.php:189]: Don't forget to remove expected classifications and cmo assets.

**Review by petertgiles — CHANGES_REQUESTED** [review]
Amost there!  Two fields left.

File: api/database/seeders/DatabaseSeeder.php (hunk 1 above)
Comment (vd1992) [reply api/database/seeders/DatabaseSeeder.php]: aye aye https://github.com/GCTC-NTGC/gc-digital-talent/pull/4266/commits/00f10ef68bf4685da7c1660f54c71b59a60c1b9c

**Review by petertgiles — APPROVED** [review]

**Pull request merged**